### ISSUE TITLE
Should we hide the ability of setting delivered status for a phase which has timeline

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/connect-app#2248]
Once we have a phase controlled via milestones, should we allow user to mark the status of phase as delivered or should we let milestones control that? IMO, later is better option and we should disable the option in the status dropdown for phase.

fyi @rmjmccormick @acshields @RishiRajSahu

**Comment by vic-topcoder** [comment]
The idea is to use the Timeline milestones flow to deliver/close the milestone. I'd prefer we do not allow switching the progress outside of the regular flow, this is confusing, as the goal is to use the milestones (progress and deliver).

The same logic applies for the phase "active" status - as soon as the first work milestone is switched as active, the phase should go into 'Active" status from 'Planned'.

Based on the milestone definition from Kyle, there is always the preparatory phase (where the copilot completes the work specification), so this one should still count for planning vs. work we're doing, but we have to confirm with Kyle (@acshields @rmjmccormick).

**Comment by rmjmccormick** [comment]
@vikasrohit @vic-topcoder Agree with both of you.

**Comment by vikasrohit** [comment]
Just fyi for

> The same logic applies for the phase "active" status - as soon as the first work milestone is switched as active, the phase should go into 'Active" status from 'Planned'.

Right now there is no option to activate a milestone in our UI. So, rather, when phase is marked as active (in progress), we mark the first milestone active.

**Comment by vic-topcoder** [comment]
That is good. Our first milestone is always closing the work specification - this is the work preparatory state that Kyle calls "requirements gathering" and said it usually lasts ~2d. The preparatory milestone doesn't really affect the work execution so if it takes a long time for the customer to reply back we don't want to be counting days of delay as progress. That is the intended design to not calculate time until preparatory milestones are closed - urges the customer to do the job of closing the specification with the Copilot.

**Comment by vikasrohit** [comment]
Didn't quite get it. Our milestones starts as soon as phase is made active and it would start  counting the milestones for the progress calculation. 

>Our first milestone is always closing the work specification

If this is added as milestone to the phase/product, it would be used for calculating the progress as soon as phase is made active. Any thing that is done before making the phase active, is not being counted for progress.

**Comment by maxceem** [comment]
Done.
For tests:
Example of a project with timelines http://local.topcoder-dev.com:3000/projects/6375/plan
Example of a project without timelines http://local.topcoder-dev.com:3000/projects/6375/plan

**Comment by vikasrohit** [comment]
@maxceem what we really intended here is to have the option for `Delivered` status but make it disabled and show tooltip just like we are doing for `duration` field. Is that possible?

**Comment by maxceem** [comment]
@vikasrohit done via https://github.com/appirio-tech/connect-app/commit/258d4786634f3a0f5e542f6ad73feddb84f140b5

**Comment by vikasrohit** [comment]
Awesome. Thanks.

**Comment by vikasrohit** [comment]
@maxceem it is not showing tooltip for me on disabled `delivered` status item.

**Comment by maxceem** [comment]
@vikasrohit oh, I've missed that requirement, I'll add a tooltip in shortly.

**Comment by vikasrohit** [comment]
Cool. Thanks. If you can add little `info` type icon (provided you find one in our already committed icons) on all such fields where we are showing such tooltip?

**Comment by maxceem** [comment]
@vikasrohit `ToolTip` component doesn't work inside `Dropdown` component, probably it will require some time to debug and fix inside `appirio-tech-react-components`, do you think we can handle it as a separate issue?

Also about adding `info` icon. Currently, disabled values are shown using disabled `input` elements, so we cannot straight add `info` sign next to the value. Will need to replace them first with regular text. As I feel like I've finished for today can I handle this tomorrow morning?

**Comment by maxceem** [comment]
@vikasrohit I've created two separate issues, as both items are non-trivial and may need some discussion:
https://github.com/appirio-tech/connect-app/issues/2338
https://github.com/appirio-tech/connect-app/issues/2339

**Comment by vikasrohit** [comment]
Agreed. Will take care both of them in next release. Please don't start working on them until we start working for next release. :)
